### PR TITLE
[2.8] Augment running hook msg for relations

### PR DESCRIPTION
**Pull request by achilleasa**
When juju appends an entry in a unit's status history, it de-dups entries with the same message and simply updates the tracked status timestamp. This makes debugging harder when working with relation hooks (e.g. relation-departed) as the hook runs for the different units get coalesced together.

This PR, augments the emitted message when running a hook to include the remote unit for relation hooks only (if defined, i.e. this is not a relation-broken hook).

## QA steps

```console
$ juju bootstrap lxd test --no-gui
$ juju deploy cs:~containers/easyrsa-345
$ juju deploy cs:~nats-charmers/nats-9 -n3
$ juju relate nats easyrsa

# Wait for units to settle

$ juju debug-hook nats/2 cluster-relation-departed

# In another terminal
$ juju remove-unit nats/2

# Wait for the *first* debug hook to trigger and press ctrl+d
# Once you get to the prompt for the *second* debug hook
# switch to another terminal and run:
$ juju show-status-log nats/2
...
running cluster-relation-departed hook for nats/0
running cluster-relation-departed hook for nats/1
```

## Bug reference
https://bugs.launchpad.net/juju/+bug/1916746

### CODE REVIEW

File: cmd/jujud/agent/caasunitinit_test.go [hunk 1]
```diff
@@ -132,7 +133,7 @@ func (s *CAASUnitInitSuite) TestInitUnit(c *gc.C) {
 }
 
 func (s *CAASUnitInitSuite) TestInitUnitWaitSend(c *gc.C) {
-	socketName := "@" + string(rand.Int63())
+	socketName := "@" + fmt.Sprint(rand.Int63())
```
Comment (SimonRichardson) [inline cmd/jujud/agent/caasunitinit_test.go:136]:
```suggestion
	socketName := fmt.Sprintf("@%d", rand.Int63())
```

**Review by SimonRichardson — APPROVED** [review]

File: cmd/jujud/agent/caasunitinit_test.go (hunk 1 above)
Comment (achilleasa) [reply cmd/jujud/agent/caasunitinit_test.go:136]: Doh! Let me rebase to fix that

**Comment by achilleasa** [comment]
!!build!! (timeout starting build)

**Comment by achilleasa** [comment]
$$merge$$

**Pull request merged**